site/compiler/test.js

```diff
@@ -133,14 +149,324 @@ describe('site/compiler', () => {
+      expect(pages[10].body).to.match(/everyone/);
+      expect(pages[11].path).to.equal('about-us/people/engineering/index.html');
+      expect(pages[11].body).to.match(/engineering/);
+      expect(pages[12].path).to.equal('about-us/people/engineering/2/index.html');
+      expect(pages[12].body).to.match(/engineering/);
+      expect(pages[13].path).to.equal('about-us/people/engineering/3/index.html');
+      expect(pages[13].body).to.match(/engineering/);
+      expect(pages[14].path).to.equal('about-us/people/leadership/index.html');
+      expect(pages[14].body).to.match(/leadership/);
+      expect(pages[15].path).to.equal('about-us/people/leadership/2/index.html');
+      expect(pages[15].body).to.match(/leadership/);
+      expect(pages[16].path).to.equal('about-us/people/leadership/3/index.html');
+      expect(pages[16].body).to.match(/leadership/);
+    });
+  });
+
+  describe('with 41 Engineers and 21 Leaderships', () => {
+    it('should render three everyone pages and three for engineering and two for leadership', () => {
+      const badgers = [];
+      for (let i = 0; i < 21; i += 1) {
+        badgers.push({
+          firstName: 'Alex ' + i,
+          lastName: 'Savin',
+          categories: [
+            'Engineering',
+            'Leadership',
+          ],
+        });
+      }
+      for (let i = 21; i < 41; i += 1) {
+        badgers.push({
+          firstName: 'Alex ' + i,
+          lastName: 'Savin',
+          categories: [
+            'Engineering',
+          ],
+        });
+      }
+      const pages = compileSite({
+        jobs: [],
+        job: {},
+        contactUsURL: '',
+        featuredBlogPosts: [],
+        events: [],
+        event: {},
+        badgers,
+        categories: ['Engineering', 'Leadership'],
+      });
+
+      expect(pages.length).to.equal(16);
+      expect(pages[8].path).to.equal('about-us/people/index.html');
+      expect(pages[8].body).to.match(/everyone/);
+      expect(pages[9].path).to.equal('about-us/people/everyone/2/index.html');
+      expect(pages[9].body).to.match(/everyone/);
+      expect(pages[10].path).to.equal('about-us/people/everyone/3/index.html');
+      expect(pages[10].body).to.match(/everyone/);
+      expect(pages[11].path).to.equal('about-us/people/engineering/index.html');
+      expect(pages[11].body).to.match(/engineering/);
+      expect(pages[12].path).to.equal('about-us/people/engineering/2/index.html');
+      expect(pages[12].body).to.match(/engineering/);
+      expect(pages[13].path).to.equal('about-us/people/engineering/3/index.html');
+      expect(pages[13].body).to.match(/engineering/);
+      expect(pages[14].path).to.equal('about-us/people/leadership/index.html');
+      expect(pages[14].body).to.match(/leadership/);
+      expect(pages[15].path).to.equal('about-us/people/leadership/2/index.html');
+      expect(pages[15].body).to.match(/leadership/);
+    });
```

Comment:
We might want to consider moving these specs to a different file, namespaced after the section of the site they actually check. It might become huge otherwise.